docs/content/3.manual/manual.yml

```diff
@@ -301,6 +301,12 @@ sections:
           to the last element, -2 referring to the next to last element,
           and so on.
 
+          The `.foo` syntax only works for simply keys i.e. keys that
+          are all alphanumeric characters. `.[<string>]` works with
+          keys that contain special charactors such as colons and dots.
+          For example `.["foo::bar"]` and `.["foo.bar"]` work while
+          `.foo::bar` and `.foo.bar` would not.
```

Comment:
The current version of the paragraph about .foo, as well as the variants that have so far been proposed, all have their problems.  I believe the following version addresses all the inaccuracies:
```
.foo, .foo.bar

Given a JSON object, the filter .foo produces the value associated with the key named "foo" if there is one;
otherwise it produces the value null.  

The .key syntax can be used provided the key name begins with an alphabetic character or an underscore, and only contains alphanumeric characters, digits, and underscores. 

If the key contains special characters, you can surround it with double quotes (e.g. ."foo$") or use the form .[s] (e.g. .["foo$"]). If s is a string-valued variable or parameter, use .[s], like this: .[$x]
```